### ISSUE TITLE
Large file uploads causing 500 error?

### ISSUE THREAD

**Issue by Jazzynupe** [git-lfs/lfs-test-server#89]
So in testing the server for using it as a folder tracking for large video files (yes I know this is not production but I need a personal folder tracker), I am so far doing well but some VERY LARGE video files are getting 500 errors as the sample below shows. Now the other situation is that I accidentally did a LARGE commit of lots of files that total 47 GB of data (141 in one commit), so there is that scenario, but only these few files show as not being successful. I am not sure if this is a server related bug or if it is related to the [Git LFS client issue with big files](https://github.com/git-lfs/git-lfs/issues/2434) as mentioned previously.

***
Fatal error: Server error: http://localhost:8080/objects/90c98ca5691b60e465b368ffcdcce249d9772f34f9a5e7da4c3affc96
2c8a021
Uploading LFS objects:  94% (132/141), 76 GB | 6.3 MB/s, done
Fatal error: Server error: http://localhost:8080/objects/a54cd24f88b6a38e10cddfd71c4356824fd58750b3f348ea9450dada5
901893f
Fatal error: Server error: http://localhost:8080/objects/2c34eb4b90dd62b40c24522563450474578fa279d8acd076e3c2d9707
9989ce7
Fatal error: Server error: http://localhost:8080/objects/ea34c8f9d1619e90955e5fa89f2e2c661ba7ca5dec1368a0cc93e7bde
b8970c8
Fatal error: Server error: http://localhost:8080/objects/5aace7e99ae1716af78f52935131644d1fb7cc6c40df1fed39ec4d5d3
f2b7d3d
Fatal error: Server error: http://localhost:8080/objects/fad2c117381850bd75e5a76aa127e5614fff44fc241f11f276d497434
7318766
Fatal error: Server error: http://localhost:8080/objects/59122b6d6dfe807b08b717af013a102594ea7e7c465c01a3474098b34
d071899
Fatal error: Server error: http://localhost:8080/objects/8ca6f92a4c4bc8e449b1dc046c3905df7cbf1596d4c8808bbc6090892
ba648e8
Fatal error: Server error: http://localhost:8080/objects/136311786a3a94961ff5cfe17388a551029608442617aae19c39c692c
4d676c8
error: failed to push some refs to 'file://localhost/GitRepo/Movies1'
***

I am trying to do some more checks but wonder if this is a bug that needed to be investigated or is it a noob problem not knowing what I am doing.

***
< HTTP/1.1 500 Internal Server Error
< Connection: close
< Content-Length: 124
< Content-Type: text/plain; charset=utf-8
< Date: Thu, 24 Oct 2019 19:52:52 GMT
<
12:52:52.287691 trace git-lfs: xfer: adapter "basic" worker 7 finished job for "ea34c8f9d1619e90955e5fa89f2e2c661ba7ca5d
ec1368a0cc93e7bdeb8970c8"
12:52:52.287691 trace git-lfs: tq: retrying object ea34c8f9d1619e90955e5fa89f2e2c661ba7ca5dec1368a0cc93e7bdeb8970c8: Fat
al error: Server error: http://localhost:8080/objects/ea34c8f9d1619e90955e5fa89f2e2c661ba7ca5dec1368a0cc93e7bdeb89
70c8
12:52:52.288692 trace git-lfs: tq: enqueue retry #2 for "ea34c8f9d1619e90955e5fa89f2e2c661ba7ca5dec1368a0cc93e7bdeb8970c
8" (size: 3294909799)
12:52:52.317693 trace git-lfs: HTTP: 500
***

I tried to read all I can but not much was published about this error or scenario.

**Comment by bk2204** [comment]
Hey,

I'm not sure exactly what's going on here, but here are some things that would help in tracking it down.  First, it's helpful if you can include the entire output of running the client with `GIT_TRACE=1`, `GIT_TRANSFER_TRACE=1`, and `GIT_CURL_VERBOSE=1`.  Those three environment variables include all of the relevant info that the client can produce.

If your files are larger than 4 GiB and you're on Windows, you're probably running into the Git problem that you linked to where it corrupts large files.  You can check to see if those files which you expect should be larger are of the appropriate size when you look at the appropriate file under `.git/lfs/objects`.  Two of the reasons you could get a 500 error are a hash mismatch or a size mismatch, and you can check the hash of those files with `sha256sum`, which is likely included with Git for Windows.

The final possibility is that a resume is going on and you have large files; it looks like the server doesn't handle 64-bit integers in `Range` headers, which I will send a patch to fix in a moment.

**Comment by Jazzynupe** [comment]
It is a VERY LARGE log. That snippit is from me running with the flags set. Both the client and server are on the same machine in this test. 

GIT_TRACE=1 GIT_TRANSFER_TRACE=1 GIT_CURL_VERBOSE=1 git push origin master

Some files are close to 4GB, but some that errored are only 1GB.
These files do show as resuming in several cases so I was going to clean up the troubled files and see if it was a size or hash issue.

Right now the details are long but can be shared at a location if needed. This is the beginning, what I posted first was toward the end where it quit.

***
User@localhost MINGW64 /q/Movies1 (master)
$ GIT_TRACE=1 GIT_TRANSFER_TRACE=1 GIT_CURL_VERBOSE=1 git push origin master
12:52:45.126790 exec-cmd.c:237          trace: resolved executable dir: C:/Program Files/Git/mingw64/bin
12:52:45.133807 git.c:440               trace: built-in: git push origin master
12:52:45.140801 run-command.c:663       trace: run_command: unset GIT_PREFIX; 'git-receive-pack '\''//localhost/Git
Repo/Movies1'\'''
12:52:45.254728 exec-cmd.c:237          trace: resolved executable dir: C:/Program Files/Git/mingw64/libexec/git-core
12:52:45.265730 git.c:440               trace: built-in: git receive-pack //localhost/GitRepo/Movies1
12:52:45.372734 run-command.c:663       trace: run_command: .git/hooks/pre-push origin file://localhost/GitRepo/Mov
ies1
12:52:45.493847 exec-cmd.c:237          trace: resolved executable dir: C:/Program Files/Git/mingw64/libexec/git-core
12:52:45.497842 git.c:703               trace: exec: git-lfs pre-push origin file://localhost/GitRepo/Movies1
12:52:45.497842 run-command.c:663       trace: run_command: git-lfs pre-push origin file://localhost/GitRepo/Movies
1
12:52:45.548844 trace git-lfs: exec: git 'version'
12:52:45.630895 trace git-lfs: exec: git '-c' 'filter.lfs.smudge=' '-c' 'filter.lfs.clean=' '-c' 'filter.lfs.process=' '
-c' 'filter.lfs.required=false' 'rev-parse' 'HEAD' '--symbolic-full-name' 'HEAD'
12:52:45.804868 trace git-lfs: exec: git 'config' '-l'
12:52:45.846840 trace git-lfs: pre-push: refs/heads/master e40981e9baf1a48ec73585201f85cb53859a0a22 refs/heads/master c5
b52604abad58ec2f2705c3b478e744c6aa27d5
12:52:46.134572 trace git-lfs: tq: running as batched queue, batch size of 100
12:52:46.135587 trace git-lfs: run_command: git rev-list --stdin --objects --not --remotes=origin --
12:52:46.789040 trace git-lfs: tq: sending batch of size 100
12:52:46.790049 trace git-lfs: api: batch 100 files
12:52:46.790049 trace git-lfs: creds: git credential fill ("http", "localhost:8080", "")
12:52:46.817126 exec-cmd.c:237          trace: resolved executable dir: C:/Program Files/Git/mingw64/libexec/git-core
12:52:46.822122 git.c:440               trace: built-in: git credential fill
12:52:46.824278 run-command.c:663       trace: run_command: 'git credential-manager get'
12:52:46.928118 exec-cmd.c:237          trace: resolved executable dir: C:/Program Files/Git/mingw64/libexec/git-core
12:52:46.932118 git.c:703               trace: exec: git-credential-manager get
12:52:46.932118 run-command.c:663       trace: run_command: git-credential-manager get
12:52:47.459232 trace git-lfs: Filled credentials for http://localhost:8080/
12:52:47.479206 trace git-lfs: HTTP: POST http://localhost:8080/objects/batch
> POST /objects/batch HTTP/1.1
> Host: localhost:8080
> Accept: application/vnd.git-lfs+json; charset=utf-8
> Authorization: Basic * * * * *
> Content-Length: 9182
> Content-Type: application/vnd.git-lfs+json; charset=utf-8
> User-Agent: git-lfs/2.8.0 (GitHub; windows amd64; go 1.12.2; git 30af66bb)
>

***

**Comment by Jazzynupe** [comment]
After some trial and error it seems it is file related. it kept failing on certain files on upload to the LFS system with the 500 error I showed. When I check the files everything is fine and the files are not corrupt or anything. So after battling it for awhile I switched to using PowerShell on windows and leveraging PoSH-Git with it.

So far the ones that appeared to have issues are working as expected and are not giving me an error. Through Powershell I am adding and committing 1 file at a time to try to narrow down the pattern but the ones that seemed to fail are successfully committing. And when I do a "git lfs ls-files" those files appear. So it could be a Bash issue on Windows. 

This is what I am doing to upload the entire set of untracked files (for posterity).

```powershell
(get-gitstatus).working | foreach-object {get-childitem $_ -recurse } | where {(-not $_.psiscontainer) -and ($_.Extension -ne ".dat" -and $_.Extension -ne ".ini")} | foreach {
$file=$_;
git add "$($file.Fullname)"
git commit -m "`"$($file.Fullname)`""
}
```
